vulnerabilities/utils.py

```diff
@@ -546,3 +547,56 @@ def get_purl_version_class(purl):
     if check_version_class:
         purl_version_class = check_version_class.version_class
     return purl_version_class
+
+
+def normalize_text(text):
+    """Normalize text by removing whitespace and converting to lowercase."""
+    return "".join(text.split()).lower() if text else ""
+
+
+def normalize_list(lst):
+    """Sort a list to ensure consistent ordering."""
+    return sorted(lst) if lst else []
+
+
+def compute_content_id(advisory_data, include_metadata=False):
+    """
+    Compute a unique content_id for an advisory by normalizing its data and hashing it.
+
+    :param advisory_data: An AdvisoryData object
+    :param include_metadata: Boolean indicating whether to include `created_by` and `url`
+    :return: SHA-256 hash digest as content_id
+    """
+
+    # Normalize fields
+    from vulnerabilities.importer import AdvisoryData
+    from vulnerabilities.models import Advisory
+
+    if isinstance(advisory_data, Advisory):
+        normalized_data = {
```

Comment:
Content id should also include `aliases` field.

Let's look at this data source https://github.com/nodejs/security-wg/blob/75c78bbbd2ef86d289c16818bb487a70e315dc43/vuln/npm/7.json. This single advisory contains 2 CVEs, and for this we create 2 different advisory records. If we do not consider aliases while computing the content ID, then we will delete the one advisory in our dedupe pipeline.